students/franjaku/lesson03/src/basic_operations.py

```diff
@@ -0,0 +1,107 @@
+"""
+    Defines the basic operations needed to operate and maintain a customer database in sqlite3.
+        -add new customers to the database
+        -search customers in the database
+        -delete existing customers from the database
+        -update customers credit
+        -list the number of active customers in the database
+
+    Uses logging to capture messages.
+"""
+
+import warnings
+import logging
+
+from customer_model import Customer, database
+
+logging.basicConfig(level=logging.ERROR)
+LOGGER = logging.getLogger(__name__)
+
+
+def add_customer(customer_id, name, last_name, home_address, phone_number, email_address, status,
+                 credit_limit):
+    """
+        Adds a new customer to the database. Throw an exception if the customer already exists.
+        Checks the input data to ensure its valid.
+        Customer ID is used as the primary  key.
+    """
+    with database.atomic():
+        try:
+            new_customer = Customer.create(customer_id=customer_id,
+                                           name=name,
+                                           last_name=last_name,
+                                           home_address=home_address,
+                                           phone_number=phone_number,
+                                           email_address=email_address,
+                                           status=status,
+                                           credit_limit=credit_limit)
+            new_customer.save()
+            LOGGER.info('New Customer: %s, created and saved to the database.', customer_id)
+        except Exception as err:
```

Comment:
Avoid using general, "catch all" exceptions. Instead, use only try / except blocks that specify the specific exceptions to be handled by the _except_ portion. This will cause any _new_ exceptions to crash your program, letting you know that previously not seen exception type has come up.